### PR TITLE
Check for the exsistance of a skel file that could be use to overrite…

### PR DESCRIPTION
**Pull request by sferich888**
… the gitconfig

This should fix https://bugzilla.redhat.com/show_bug.cgi?id=1265423

### CODE REVIEW

File: node/lib/openshift-origin-node/model/application_repository.rb [hunk 1]
```diff
@@ -262,9 +262,10 @@ def configure
           File.open(f, 'w', m) { |f| f.write(ERB.new(t).result(binding)) }
         }
 
-        render_file.call(PathUtils.join(@path, 'description'), 0644, GIT_DESCRIPTION)
-        render_file.call(PathUtils.join(@container.container_dir, '.gitconfig'), 0644, GIT_CONFIG)
-
+        render_file.call(PathUtils.join(@path, 'description'), 0644, GIT_DESCRIPTION
```
Comment (tiwillia) [inline node/lib/openshift-origin-node/model/application_repository.rb]: Missing parenthesis at the end here. Looks like it was moved to the line below.

**Comment by tiwillia** [comment]
The `[pack]` section of the hard-coded gitconfig is important and should always be included. It was added in commit 47065697e2d01521ae650a34ee4fa7c22d49fd26 due to the issue described here:
  http://stackoverflow.com/questions/4826639/repack-of-git-repository-fails

Perhaps we can devise a way to append to the gitconfig, rather than completely overwrite it?

**Comment by sferich888** [comment]
@tiwillia I think that can be solved with documentation! The use case for this should be limited to people looking to customize values in the config, often the [pack] settings. 

In short "if providing your own .gitconfig you need to ensure you provide a [pack] section to avoid said issues".

**Comment by sferich888** [comment]
@tiwillia I should have fixed the issue with the parenthesis

**Comment by tiwillia** [comment]
@sferich888 There are 3 commits, you'll need to squash them.

**Comment by tiwillia** [comment]
LGTM as long as we make sure docs are updated with instructions to include the `[pack]` and `[gc]` sections of the original `.gitconfig`.

@Miciah could you give this a quick review and merge? I'd appreciate a second set of eyes on this.

**Comment by Miciah** [comment]
Looks reasonable... but I wonder whether it would be worth adding a test case.

**Comment by abhgupta** [comment]
Are we waiting on test cases to be added for this?

**Comment by sferich888** [comment]
@Miciah @abhgupta what would we be testing?  If a file exists? Shouldent the core of ruby be providing the tests for that function.

**Comment by abhgupta** [comment]
@Miciah this doesn't seem to require any unit tests - go ahead with the merge if there is nothing else blocking this PR

**Comment by abhgupta** [comment]
@sferich888 @tiwillia make sure we track the documentation for this change.

**Comment by Miciah** [comment]
openshift-bot, please [merge]!

**Comment by abhgupta** [comment]
@sferich888 @tiwillia Was a doc note added for this?

**Comment by sferich888** [comment]
@abhgupta Docs changes are needed, and would be limited to: 

1) The .gitconfig file can now be overwritten by a .gitconfig file provided in the user's git repostiroy
2) The user-provided .gitconfig should contain the following options to ensure the git repository can be cloned with the gear is close to its memory limits:

	[gc]
	  auto=100
	[pack]
	  windowMemory = 10m
	  packSizeLimit = 20m

**Comment by abhgupta** [comment]
@tiwillia @Miciah can one of you reach out to Docs to make sure this is captured?

**Comment by Miciah** [comment]
I'm following up in the Bugzilla report. which I'll clone to track the required changes to documentation.

**Pull request merged**